core-web/libs/sdk/react/src/lib/components/DotEditableText/DotEditableText.tsx

```diff
@@ -0,0 +1,141 @@
+    const [content, setContent] = useState(contentlet[fieldName] || '');
+
+    useEffect(() => {
+        setisInsideEditor(isInsideEditorFn());
+
+        if (!isInsideEditor) {
+            return;
+        }
+    }, [isInsideEditor]);
+
+    useEffect(() => {
+        if (!isInsideEditor) {
+            return;
+        }
+
+        const content = contentlet[fieldName] || '';
+        editorRef.current?.setContent(content, { format });
+        setContent(content);
+    }, [contentlet, fieldName, format, isInsideEditor]);
+
+    useEffect(() => {
+        const onMessage = ({ data }: MessageEvent) => {
+            const { name, payload } = data;
+            if (name !== 'COPY_CONTENTLET_INLINE_EDITING_SUCCESS') {
+                return;
+            }
+
+            const { oldInode, inode } = payload;
+            const currentInode = contentlet.inode;
+
+            if (currentInode === oldInode || currentInode === inode) {
+                editorRef.current?.focus();
+
+                return;
+            }
+        };
+
+        window.addEventListener('message', onMessage);
+
+        return () => {
+            window.removeEventListener('message', onMessage);
+        };
+    }, [contentlet.inode]);
+
+    const onMouseDown = (event: MouseEvent) => {
+        const { onNumberOfPages = 1 } = contentlet;
+
+        if (onNumberOfPages <= 1 || editorRef.current?.hasFocus()) {
+            return;
+        }
+
+        const { inode, languageId: language } = contentlet;
+
+        event.stopPropagation();
+        event.preventDefault();
+
+        try {
+            postMessageToEditor({
+                action: CUSTOMER_ACTIONS.COPY_CONTENTLET_INLINE_EDITING,
+                payload: {
+                    dataset: {
+                        inode,
+                        language,
+                        fieldName
+                    }
+                }
+            });
+        } catch (error) {
+            console.error('Failed to post message to editor:', error);
+        }
+    };
+
+    const onFocusOut = () => {
+        const editedContent = editorRef.current?.getContent({ format: format }) || '';
+
+        if (!editorRef.current?.isDirty() || !didContentChange(editedContent)) {
+            return;
+        }
+
+        const { inode, languageId: langId } = contentlet;
+
+        try {
+            postMessageToEditor({
+                action: CUSTOMER_ACTIONS.UPDATE_CONTENTLET_INLINE_EDITING,
+                payload: {
+                    content: editedContent,
+                    dataset: {
+                        inode,
+                        langId,
+                        fieldName
+                    }
+                }
+            });
+        } catch (error) {
+            console.error('Failed to post message to editor:', error);
+        }
+    };
+
+    const didContentChange = (editedContent: string) => {
+        return content !== editedContent;
+    };
```

Comment:
I wonder how similar is this from the Angular one. Maybe we can create a shared utils for this in the client lib. 

It can be in another iteration, I just want to know if it's possible

Reply:
I can create a util to unify them. Where can these utils live? Under `libs/sdks/utils`?

cc: @fmontes @KevinDavilaDotCMS

Reply:
There are buildable libraries so we can't use absolute paths like `'@dotcms/utils'`

Reply:
I agree with the `libs/sdks/utils` idea. I think we already have something like that

Reply:
@zJaaal

I'm getting this error, this should be possible when we create the @utils and @types packages

```bash
Projects cannot be imported by a relative or absolute path, and must begin with a npm scopeeslint[@nx/enforce-module-boundaries](https://github.com/nrwl/nx/blob/19.5.6/docs/generated/packages/eslint-plugin/documents/enforce-module-boundaries.md)
(alias) type DOT_EDITABLE_TEXT_MODE = "minimal" | "full" | "plain"
import DOT_EDITABLE_TEXT_MODE
```

![Error importing something outside the lib](https://github.com/user-attachments/assets/fc0033b5-fc32-4f4b-9fe3-bf67ea8f1515)